apis/addons/v1alpha1/addon_webhook.go

```diff
@@ -0,0 +1,42 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var addonlog = logf.Log.WithName("addon-resource")
+
+func (r *Addon) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+/*
+This marker is responsible for generating a validating webhook manifest.
+*/
+
+//+kubebuilder:webhook:verbs=create;update;delete,path=/validate-addon,mutating=false,failurePolicy=fail,groups=addons.managed.openshift.io,resources=addons,versions=v1alpha1,name=addons.managed.openshift.io,sideEffects=None,admissionReviewVersions=v1
```

Comment:
How do we make kubebuilder generate this manifest? I thought running `make generate` would do it 🤔